base/linalg/exceptions.jl

```diff
@@ -29,10 +29,29 @@ function Base.showerror(io::IO, ex::ARPACKException)
     end
 end
 
+"""
+   SingularException(info)
+
+The matrix passed is a singular matrix. Optional argument INFO is an INTEGER
+= 0:  successful exit
+< 0:  if INFO = -i, the i-th argument had an illegal value
+> 0:  if INFO = i, U(i,i) is exactly zero. The factorization
+      has been completed, but the factor U is exactly
```

Comment:
`U` is specific to the decomposition, this exception occurs in cases where there is no `U`.